### PR TITLE
Fix #44 Maven CI build issue

### LINKED ISSUE #44: Bug - Fix the github action maven CI build failure
The current github action for maven CI build is failing everytime because of `TestHttpRESTApi` class which is using a `MockServer` for testing .

So , need to check how this issue can be fixed

### PR DESCRIPTION
**Pull request by thanhtamB2112004**
Update some alters about dependencies and plug-ins for Maven build.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -91,35 +90,28 @@
...
+                <configuration>
+                    <source>21</source>
+                    <target>21</target>
+                </configuration>
+            </plugin>
 			<plugin>
 				<groupId>org.apache.maven.plugins</groupId>
 				<artifactId>maven-source-plugin</artifactId>
 				<version>3.3.1</version>
-				<executions>
```
Comment (animo93) [inline pom.xml:115]: Why is the execution block from maven-source-plugin removed ?

File: pom.xml [hunk 2]
```diff
@@ -15,15 +15,14 @@
 		<project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 		<maven.compiler.source>21</maven.compiler.source>
 		<maven.compiler.target>21</maven.compiler.target>
-		<maven.compiler.release>21</maven.compiler.release>
```
Comment (animo93) [inline pom.xml:18]: Why was this removed ?

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (animo93) [inline .vscode/settings.json:1]: We never commit project files in git . Please add this to your .gitignore

**Review by animo93 — CHANGES_REQUESTED** [review]
Please do work on the suggested changes

File: pom.xml (hunk 1 above)
Comment (thanhtamB2112004) [reply pom.xml:115]: Sorry, I made a mistake here when altering the plugins, I will fix this in the commit.

File: pom.xml (hunk 2 above)
Comment (thanhtamB2112004) [reply pom.xml:18]: I thought the release contains the functioning of both source and target, if it serves a different task I will fix it.

File: .vscode/settings.json (hunk 3 above)
Comment (thanhtamB2112004) [reply .vscode/settings.json:1]: Yes, I added it.

File: .github/workflows/codeql-analysis.yml.disable
[the remark this answers is not in the document]
Comment (thanhtamB2112004) [reply .github/workflows/codeql-analysis.yml.disable:1]: Yes, I have changed it.

**Review by animo93 — APPROVED** [review]
Approved
